pkg/agent/client_test.go

```diff
@@ -129,6 +129,9 @@ func TestServeData_HTTP(t *testing.T) {
 		t.Errorf("expect nil closeErr; got %v", closeErr)
 	}
 
+	// give agent time to clean up.
+	time.Sleep(250 * time.Millisecond)
```

Comment:
I strongly dislike sleeps in tests. There is always a tradeoff between test performance & flakiness. In a heavily contended CI environment, it's possible that a process could be paused for multiple seconds, which basically ignores the sleep (since it's wall time).

When it's not possible to get a definitive "done" signal, it's better to poll for the desired state. Fortunately we already have a helper to do just that: `waitForConnectionDeletion(t, testClient, connID)`